samples/components/dialects/Hive12Dialect.tdd

```diff
@@ -1100,106 +1158,174 @@
+      <formula part='day'>DAY(%1)</formula>
+      <formula part='weekday'>(8 + DATEDIFF(%1,NEXT_DAY(CAST(%1 AS DATE),&apos;SU&apos;)))</formula>
+      <formula part='week'>FLOOR((14 + DATEDIFF(%1, TRUNC(%1,&apos;YY&apos;)) + DATEDIFF(TRUNC(%1,&apos;YY&apos;),NEXT_DAY(TRUNC(%1,&apos;YY&apos;),&apos;SU&apos;)))/7)</formula>
+      <formula part='hour'>HOUR(%1)</formula>
+      <formula part='minute'>MINUTE(%1)</formula>
+      <formula part='second'>SECOND(%1)</formula>
+      <argument type='localstr' />
+      <argument type='datetime' />
+      <argument type='localstr' />
+    </date-function>
+    <date-function name='DATETRUNC' return-type='datetime'>
+      <formula part='year'>CAST(TRUNC(%1,&apos;YY&apos;) AS DATE)</formula>
+      <formula part='quarter'>CAST(CONCAT(YEAR(%1), (CASE WHEN MONTH(%1)&lt;4 THEN &apos;-01&apos;  WHEN MONTH(%1)&lt;7 THEN &apos;-04&apos;  WHEN MONTH(%1)&lt;10 THEN &apos;-07&apos;  ELSE &apos;-10&apos; END), &apos;-01 00:00:00&apos;) AS TIMESTAMP)</formula>
+      <formula part='month'>CAST(TRUNC(%1,&apos;MM&apos;) AS DATE)</formula>
+      <formula part='dayofyear'>CAST (%1 AS DATE)</formula>
+      <formula part='day'>CAST(TO_DATE(%1) AS DATE)</formula>
+      <formula part='weekday'>CAST (%1 AS DATE)</formula>
+      <formula part='week'>CAST(DATE_ADD(NEXT_DAY(sunday,&apos;SU&apos;),-7) AS DATE)</formula>
+      <formula part='hour'>CAST(FROM_UNIXTIME(UNIX_TIMESTAMP(%1), &apos;yyyy-MM-dd HH:00:00&apos;) AS TIMESTAMP)</formula>
+      <formula part='minute'>CAST(FROM_UNIXTIME(UNIX_TIMESTAMP(%1), &apos;yyyy-MM-dd HH:mm:00&apos;) AS TIMESTAMP)</formula>
+      <formula part='second'>%1</formula>
+      <argument type='localstr' />
+      <argument type='datetime' />
+    </date-function>
+    <date-function name='DATETRUNC' return-type='datetime'>
+      <formula part='year'>CAST(TRUNC(%1,&apos;YY&apos;) AS DATE)</formula>
+      <formula part='quarter'>CAST(CONCAT(YEAR(%1), (CASE WHEN MONTH(%1)&lt;4 THEN &apos;-01&apos;  WHEN MONTH(%1)&lt;7 THEN &apos;-04&apos;  WHEN MONTH(%1)&lt;10 THEN &apos;-07&apos;  ELSE &apos;-10&apos; END), &apos;-01 00:00:00&apos;) AS TIMESTAMP)</formula>
+      <formula part='month'>CAST(TRUNC(%1,&apos;MM&apos;) AS DATE)</formula>
+      <formula part='dayofyear'>CAST (%1 AS DATE)</formula>
+      <formula part='day'>CAST(TO_DATE(%1) AS DATE)</formula>
+      <formula part='weekday'>CAST (%1 AS DATE)</formula>
+      <formula part='week'>CAST(DATE_ADD(NEXT_DAY(%1,&apos;SU&apos;),-7) AS DATE)</formula>
+      <formula part='hour'>CAST(FROM_UNIXTIME(UNIX_TIMESTAMP(%1), &apos;yyyy-MM-dd HH:00:00&apos;) AS TIMESTAMP)</formula>
+      <formula part='minute'>CAST(FROM_UNIXTIME(UNIX_TIMESTAMP(%1), &apos;yyyy-MM-dd HH:mm:00&apos;) AS TIMESTAMP)</formula>
+      <formula part='second'>%1</formula>
       <argument type='localstr' />
       <argument type='datetime' />
       <argument type='localstr' />
     </date-function>
   </function-map>
```

Comment:
It seems like you'll need more here too.  For example in SELECT, by default TOP is used, but the Hive dialects use LIMIT.